haystack/dataclasses/chat_message.py

```diff
@@ -30,8 +39,11 @@ class ChatMessage:
     content: str
     role: ChatRole
     name: Optional[str]
+    tool_call_id: Optional[str]
+    tool_calls: List[ToolCall] = field(default_factory=list)
     meta: Dict[str, Any] = field(default_factory=dict, hash=False)
 
+    # this should probably be moved to OpenAI specific classes
```

Comment:
Indeed. The general idea is to have code in each generator that parses its specific tool call into a tuple that contains the `Tool` instance that needs to be invoked and a dict of parameters.